src/Features/CustomerEffortScoreTracks.php

```diff
@@ -69,27 +84,70 @@ private function enable_survey_enqueing_if_tracking_is_enabled() {
 			return;
 		}
 
-		add_action(
-			'admin_init',
-			array(
-				$this,
-				'maybe_clear_ces_tracks_queue',
-			)
-		);
-
-		add_action(
-			'woocommerce_update_options',
-			array(
-				$this,
-				'run_on_update_options',
-			),
-			10,
-			3
-		);
+		add_action( 'admin_init', array( $this, 'maybe_clear_ces_tracks_queue' ) );
+		add_action( 'woocommerce_update_options', array( $this, 'run_on_update_options' ), 10, 3 );
+		add_action( 'product_cat_add_form', array( $this, 'add_script_track_product_categories' ), 10, 3 );
+		add_action( 'product_tag_add_form', array( $this, 'add_script_track_product_tags' ), 10, 3 );
+		add_action( 'woocommerce_attribute_added', array( $this, 'run_on_add_product_attributes' ), 10, 3 );
 
 		$this->onsubmit_label = __( 'Thank you for your feedback!', 'woocommerce-admin' );
 	}
 
+	/**
+	 * Returns a generated script for tracking tags added on edit-tags.php page.
+	 * CES survey is triggered via direct access to wc/customer-effort-score store
+	 * via wp.data.dispatch method.
+	 *
+	 * Due to lack of options to directly hook ourselves into the ajax post request
+	 * initiated by edit-tags.php page, we infer a successful request by looking
+	 * for a new tag ID generated within tags table.
+	 *
+	 * @param string $action Action name for the survey.
+	 * @param string $label Label for the snackbar.
+	 * @param string $page_now Value of window.pagenow.
+	 * @param string $admin_page Value of window.adminpage.
+	 *
+	 * @return string Generated JavaScript to append to page.
+	 */
+	private function get_script_track_edit_php( $action, $label, $page_now, $admin_page ) {
+		return sprintf(
+			"(function( $ ) {
+				'use strict';
+				// Returns an array of tag IDs based on the table displayed.
+				function getTagIds() {
+					return $('.tags tbody > tr').map( function( idx, tr ) { return tr.getAttribute( 'id' ).split( '-' )[ 1 ]; } ).toArray();
+				}
+				// Gets the highest tag ID - assuming it's the latest.
+				function getMaxId() {
+					return Math.max.apply( this, getTagIds() );
+				}
+				// Hook on submit button and sets a 500ms interval function
+				// to determine successful add tag or otherwise.
+				$('#addtag #submit').on( 'click', function() {
+					const lastId = getMaxId();
```

Comment:
Is it possible the most recently added tag won't have the highest ID? For example, if an old tag is added after a  new one is created.

Reply:
@adrianduffell I'm not sure if I follow, how will an old tag be added after a new one? 🤔

Reply:
Sorry that was confusing 🙂 

I mean when an old tag (existing on their site for a long time) is added to the product. It's going to have a small ID so I'm not sure if `getMaxId()`, if the product already has newer tags added. I'm probably missing something!

Reply:
I don't think adding an old tag to a product will make a difference since the tags table that's currently displayed is not going to change. However, this is no longer relevant since we're no longer using ID, thanks for the suggestion @adrianduffell!